lxd/container_lxc.go

```diff
@@ -2489,54 +2505,63 @@ func (c *containerLXC) startCommon() (string, error) {
 					}
 					err := c.setupUnixDevice(fmt.Sprintf("unix.%s", k), m, gpu.major, gpu.minor, gpu.path, true, false)
 					if err != nil {
-						return "", err
+						return "", postStartHooks, err
 					}
 				}
 			}
 
 			if !found {
 				msg := "Failed to detect requested GPU device"
 				logger.Error(msg)
-				return "", fmt.Errorf(msg)
+				return "", postStartHooks, fmt.Errorf(msg)
 			}
 		} else if m["type"] == "disk" {
 			if m["path"] != "/" {
 				diskDevices[k] = m
 			}
-		} else if m["type"] == "nic" || m["type"] == "infiniband" {
+		} else if m["type"] == "nic" || m["type"] == "infiniband" || m["type"] == "proxy" {
 			networkKeyPrefix := "lxc.net"
 			if !util.RuntimeLiblxcVersionAtLeast(2, 1, 0) {
 				networkKeyPrefix = "lxc.network"
 			}
 
-			m, err := c.fillNetworkDevice(k, m)
-			if err != nil {
-				return "", err
+			// Populate network nic name and hwaddr.
+			if m["type"] == "nic" || m["type"] == "infiniband" {
+				m, err = c.fillNetworkDevice(k, m)
+				if err != nil {
+					return "", postStartHooks, err
+				}
 			}
 
 			// Use new Device interface if supported.
 			nicID++
```

Comment:
Doing the proxy stuff in there may cause gaps in network interface indices if I'm reading this right.

Reply:
Fixed